compiler/rustc_target/src/asm/loongarch64.rs

```diff
@@ -0,0 +1,131 @@
+use super::{InlineAsmArch, InlineAsmType};
+use rustc_macros::HashStable_Generic;
+use rustc_span::Symbol;
+use std::fmt;
+
+def_reg_class! {
+    LoongArch64 LoongArch64InlineAsmRegClass {
+        reg,
+        freg,
+    }
+}
+
+impl LoongArch64InlineAsmRegClass {
+    pub fn valid_modifiers(self, _arch: super::InlineAsmArch) -> &'static [char] {
+        &[]
+    }
+
+    pub fn suggest_class(self, _arch: InlineAsmArch, _ty: InlineAsmType) -> Option<Self> {
+        None
+    }
+
+    pub fn suggest_modifier(
+        self,
+        _arch: InlineAsmArch,
+        _ty: InlineAsmType,
+    ) -> Option<(char, &'static str)> {
+        None
+    }
+
+    pub fn default_modifier(self, _arch: InlineAsmArch) -> Option<(char, &'static str)> {
+        None
+    }
+
+    pub fn supported_types(
+        self,
+        arch: InlineAsmArch,
+    ) -> &'static [(InlineAsmType, Option<Symbol>)] {
+        match self {
+            Self::reg => types! { _: I8, I16, I32, I64, F32, F64; },
+            Self::freg => types! { _: F32, F64; },
+        }
+    }
+}
+
+def_regs! {
+    LoongArch64 LoongArch64InlineAsmReg LoongArch64InlineAsmRegClass {
+        r6: reg = ["$r6","a2"],
+        r7: reg = ["$r7","a3"],
+        r8: reg = ["$r8","a4"],
+        r9: reg = ["$r9","a5"],
+        r10: reg = ["$r10","a6"],
+        r11: reg = ["$r11","a7"],
+        r12: reg = ["$r12","t0"],
+        r13: reg = ["$r13","t1"],
+        r14: reg = ["$r14","t2"],
+        r15: reg = ["$r15","t3"],
+        r16: reg = ["$r16","t4"],
+        r17: reg = ["$r17","t5"],
+        r18: reg = ["$r18","t6"],
+        r19: reg = ["$r19","t7"],
+        r20: reg = ["$r20","t8"],
+        r23: reg = ["$r23","s0"],
+        r24: reg = ["$r24","s1"],
+        r25: reg = ["$r25","s2"],
+        r26: reg = ["$r26","s3"],
+        r27: reg = ["$r27","s4"],
+        r28: reg = ["$r28","s5"],
+        r29: reg = ["$r29","s6"],
+        r30: reg = ["$r30","s7"],
+        r31: reg = ["$r31","s8"],
+        f0: freg = ["$f0","fa0","fv0"],
+        f1: freg = ["$f1","fa1","fv1"],
```

Comment:
As @xry111 said, it's better to add inline asm support in a later PR. Let's focus on basic arch bring-up here.

Also please remove the `fv*` aliases. According to [the ELF psABI spec](https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html), Rust belongs to the "components not yet upstream, and not interacting with other components that may expect such usage" case, hence support for the `v*` aliases is to be never upstreamed.